networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/network/NetworkConfig.java

```diff
@@ -87,10 +89,24 @@ public NetworkConfig(
     this.advertisedIp = advertisedIp.filter(ip -> !ip.isBlank());
     this.targetSubnetSubscriberCount = targetSubnetSubscriberCount;
     if (this.advertisedIp.map(ip -> !isInetAddress(ip)).orElse(false)) {
-      throw new IllegalArgumentException("Advertised ip is set incorrectly.");
+      throw new InvalidConfigurationException(
+          String.format(
+              "Advertised ip (%s) is set incorrectly.", this.advertisedIp.orElse("EMPTY")));
     }
 
-    this.listenPort = listenPort;
+    if (listenPort == 0) {
+      this.listenPort = listenPort;
+    } else {
+      if (PortAvailability.isPortAvailable(listenPort)) {
+        this.listenPort = listenPort;
+      } else {
+        throw new InvalidConfigurationException(
+            String.format(
+                "P2P Port %d (TCP/UDP) is already in use. "
+                    + "Check for other processes using this port.",
+                listenPort));
+      }
+    }
```

Comment:
Also this will check if the port is in use at startup, but we need to also specifically handle when the port was free at startup and became in use before we actually opened the socket.  This would happen if you were tracking a chain prior to the genesis event being known and running either two Teku's or a Teku and a Lighthouse - both would use port 9000 for p2p by default and only start listening when genesis becomes known (after startup).